code/game/objects/effects/decals/Cleanable/fuel.dm

```diff
@@ -90,6 +96,26 @@
 			INVOKE_NEXT_TICK(other, .proc/ignite_fuel)	//Spread effect
 	qdel(src)
 
+///Ignites when something hot enters our loc, either via crossed or signal wrapper
+/obj/effect/decal/cleanable/liquid_fuel/proc/check_ignite(atom/movable/igniter)
+	if(isitem(igniter))
+		var/obj/item/ignitem = igniter
+		if(ignitem.damtype != BURN)
+			return
+	else if(isliving(igniter))
+		var/mob/living/burner_mob = igniter
+		if(!burner_mob.on_fire)
+			return
+	else
+		return
+	visible_message(span_warning("The spilled fuel catches fire!"))
+	ignite_fuel()
+
+///Wrapper for ignition via signals rather than from crossed
+/obj/effect/decal/cleanable/liquid_fuel/proc/ignite_check_wrapper(signal_source, atom/movable/igniter)
```

Comment:
you don't really need that wrapper, you could do check_ignite(null, AM) line 42 and have check ignite be the signal_handler.

This is fine though, just for future reference